src/palette/internal/palettedraw.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "palettedraw.h"
+
+#include "draw/painter.h"
+
+#include "libmscore/accidental.h"
+
+#include "libmscore/note.h"
+
+#include "infrastructure/rtti.h"
+
+using namespace mu::palette;
+using namespace mu::engraving;
+using namespace mu::engraving::rtti;
+
+void PaletteDraw::drawItem(const EngravingItem* item, draw::Painter* painter)
+{
+    switch (item->type()) {
+    case ElementType::ACCIDENTAL:   draw(item_cast<const Accidental*>(item), painter);
+        break;
+    default:
+        item->draw(painter);
+    }
+}
+
+void PaletteDraw::draw(const Accidental* item, draw::Painter* painter)
+{
+    // don't show accidentals for tab or slash notation
+    if (item->onTabStaff() || (item->note() && item->note()->fixed())) {
+        return;
+    }
```

Comment:
(This is not strictly necessary in the Palettes, of course; but perhaps we should just focus on moving code for now, and do refactoring later.)